lib/improver/utilities/cube_manipulation.py

```diff
@@ -685,3 +687,125 @@ def sort_coord_in_cube(cube, coord, order="ascending"):
+        raise_exception (bool):
+            Option as to whether an exception should be raised, if the
+            requested coordinate is not present.
+
+    Returns:
+        cube (iris.cube.Cube):
+            Cube where the requirement for the dimensions to be in a particular
+            order will have been enforced.
+
+    Raises:
+        ValueError: The anchor argument must have a value of either start or
+            end.
+        CoordinateNotFoundError: The requested coordinate is not available on
+            the cube.
+        ValueError: Multiple coordinates match the partial name provided.
+
+    """
+    if isinstance(coord_names, unicode):
+        coord_names = str(coord_names)
+    if isinstance(coord_names, str):
+        coord_names = [coord_names]
+
+    if anchor not in ["start", "end"]:
+        msg = ("The value for the anchor must be either 'start' or 'end'."
+               "The value specified for the anchor was {}".format(anchor))
+        raise ValueError(msg)
+
+    # Determine coordinate indices for use in creating a dictionary.
+    # These indices are either relative to the start or end of the available
+    # dimension coordinates.
+    coord_indices = np.array(range(len(coord_names)))
+    if anchor == "end":
+        coord_indices = sorted(len(cube.dim_coords) - coord_indices)
+    coord_dict = dict(zip(coord_names, coord_indices))
+
+    for coord_name in coord_dict.keys():
+        # Deal with the coord_name being a partial match to the actual
+        # coordinate name.
+        if cube.coords(coord_name):
+            full_coord_name = coord_name
+        else:
+            coord = [c for c in cube.coords() if coord_name in c.name()]
+            # If the coordinate is desired, raise an exception
+            # if the coordinate is missing.
+            if len(coord) == 0:
+                if raise_exception:
+                    msg = ("The requested coordinate {} is not a coordinate "
+                           "in the cube: {}".format(coord, cube))
+                    raise CoordinateNotFoundError(msg)
+                else:
+                    continue
+            elif len(coord) == 1:
+                # Replace the dictionary key with the actual coordinate name.
+                full_coord_name = coord[0].name()
+                coord_dict[full_coord_name] = coord_dict.pop(coord_name)
+            else:
+                msg = ("More than 1 coordinate: {} matched the specified "
+                       "coordinate name: {}. Unable to distinguish which "
+                       "coordinate should be reordered.".format(
+                           coord, coord_name))
+                raise ValueError(msg)
+
+        # If the requested coordinate is not a dimension coordinate, make it
+        # a dimension coordinate.
+        if cube.coords(full_coord_name, dim_coords=False):
+            if promote_scalar:
+                cube = iris.util.new_axis(cube, full_coord_name)
```

Comment:
Note that the behaviour of the plugin differs if promote_scalar=True. This promotes the scalar coordinate in the output cube, but the input cube remains unchanged; when this is false the input cube is reordered. I'm not sure this matters, but it means at this point we duplicate the cube.

Reply:
I've added this in a comment.